### PR TITLE
add new plugin: voice cooperation

### PR DESCRIPTION
**Pull request by sparrowhe**

### CODE REVIEW

File: src/plugins/voice-cooperation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      .on(LiveKit.RoomEvent.TrackSubscribed, handleNewTrack)
+      .on(LiveKit.RoomEvent.TrackUnsubscribed, handleTrackUnsubscribed)
+      .on(LiveKit.RoomEvent.Reconnected, () => handleReconnect(room))
+      .on(LiveKit.RoomEvent.TrackMuted, () => fetchCurrentUserList(room))
+      .on(LiveKit.RoomEvent.TrackUnmuted, () => fetchCurrentUserList(room))
+      .on(LiveKit.RoomEvent.ParticipantConnected, () => newTeamMember.play());
+  };
+
+  const handleNewTrack = (track: LiveKit.RemoteAudioTrack) => {
+    console.log("new track", track);
```
Comment (zxq142857) [inline src/plugins/voice-cooperation/index.tsx:167]: These debugging codes should be removed

File: src/plugins/voice-cooperation/index.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+import VoiceIcon from "assets/icon--voice.svg";
+import { Button, GandiProvider } from "@gandi-ide/gandi-ui";
+import { connectToRoom } from "./lib/livekit";
+import { Member } from "./components/MemberList/MemberListItem";
+import toast from "react-hot-toast";
+import classNames from "classnames";
+import VoiceFloating from "./components/VoiceFloating/VoiceFloating";
+import LeaveCallIcon from "assets/icon--voice--off-white.svg";
+import { IntlShape } from "react-intl";
+import { AxiosResponse } from "axios";
```
Comment (zxq142857) [inline src/plugins/voice-cooperation/index.tsx:14]: This isn't being used anymore

File: src/plugins/voice-cooperation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+interface ITokenRequest {
+  clientId: string;
+  creationId: string;
+  authority: string;
+}
+
+interface IResponse<T> {
```
Comment (zxq142857) [inline src/plugins/voice-cooperation/index.tsx:22]: This isn't being used anymore

File: src/plugins/voice-cooperation/config.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// function getServerUrlFromQuery() {
+//   const url = new URL(window.location.href);
+//   return decodeURI(url.searchParams.get("voiceServer"));
+// }
+
+export default {
+  SERVER_URL: "wss://webrtc-test.ccw.site",
```
[the remark this answers is not in the document]
Comment (sparrowhe) [reply src/plugins/voice-cooperation/config.ts:7]: affirm

**Pull request merged**